tests/scripts/mbedtls_test.py

```diff
@@ -310,7 +309,7 @@ def run_test(self, name, function_id, dependencies, args):
 
         param_bytes, length = self.test_vector_to_bytes(function_id,
                                                         dependencies, args)
-        self.send_kv(length, param_bytes)
+        self.send_kv(''.join('{:02x}'.format(x) for x in length), ''.join('{:02x}'.format(x) for x in param_bytes))
```

Comment:
.decode() doesn't work on both pythons?

Reply:
It does, but results in different output

Reply:
The problem happens when sending a hex string in the test.
The decode tries to translate it to a string, and sends wrong data
For example:
~~~
>>> data=b'\x00\x11\x22\x33\x44'
>>> print(data.decode())
"3D
~~~
The target doesn't translate this information correct.
This is on both pythons